prescient/engine/forecast_helper.py

```diff
@@ -33,6 +49,60 @@ def get_forecastables(*models: EgretModel) -> Iterable[ Tuple[MutableSequence[fl
         yield tuple(m.data['elements']['load'][bus]['p_load']['values'] for m in models)
 
     # Reserve requirement
-    yield tuple(m.data['system']['reserve_requirement']['values'] for m in models)
+    if 'reserve_requirement' in model1.data['system']:
+        yield tuple(m.data['system']['reserve_requirement']['values'] for m in models)
```

Comment:
I'm more than little confused by the interface for this generator. Maybe not for this PR.

Reply:
Do you mean just the reserve requirement part or the whole generator function?

If you mean the function as a whole, it was created so calling code can work with forecast/actuals data without needing to know exactly what the specific data items are.  If we change what is included in a forecast (such as new kinds of reserve requirements), we can change it here and everywhere else automatically picks up the new data items.

To use the function, pass any number of models to the function and it yields tuples, where each tuple has one timeseries value array from each passed in model.  The data in any given tuple is the same "spot" in the model.  The change you have highlighted here returns a tuple where each item in the tuple is the reserve_requirement array for each model.

Note that we can safely assume that all the models have the same general layout (same generators, same presence or absence of reserve requirements) because that's pretty much a requirement for Prescient already. Here's an example of working with the function:

```
# Assume m1, m2, and m3 are all egret models

for vals in get_forecastables(m1, m2, m3):
    # vals is a 3-tuple, where each element is an array of values
    vals1 = vals[0]
    vals2 = vals[1]
    vals3 = vals[3]
    # vals1 holds a value array from m1
    # vals2 holds a value array from m2
    # vals3 holds a value array from m3
   # All three value arrays are for the same "thing", such as the same generator's p_min

    # The first time through the for loop, vals happens to be the first generator's p_min,
    # The second time through the loop, vals happens to be the first generator's p_max,
    # The third time through the loop, vals happens to be the second generator's p_min,
    # and so on. I say "happens to be" because the calling code doesn't know or care 
    # what the interpretation of each value array is, just that all arrays in the same tuple 
    # correspond to each other (they are all p_min for the same generator, for example).

# This is how you use the function with a single model
for (vals,) in get_forecastables(m1):
    # vals holds an array of values from m1.
    print(vals)

```

Reply:
This all make sense -- the original context that got me here was the later one. For an internal interface it definitely works, just confusing to me at first glace.